ethereumj-core/src/main/java/org/ethereum/vm/DataWord.java

```diff
@@ -184,158 +215,153 @@ public boolean isNegative() {
         return result == 0x80;
     }
 
-    public DataWord and(DataWord w2) {
-
+    public DataWord and(DataWord word) {
+        byte[] newData = this.getData();
```

Comment:
It's very confusing to use `getData()` for making an array copy. It better to make it explicitly or add `copyData()` private method
The similar below.